Src/WitsmlExplorer.Frontend/components/ContentViews/Charts/LogsGraph.tsx

```diff
@@ -0,0 +1,307 @@
+    cartesianXBounds[1] = retypedSys.width;
+    cartesianYBounds[0] = retypedSys.y;
+    cartesianYBounds[1] = retypedSys.height;
+    let barLength = end[0] - start[0];
+
+    //assures minimum bar lenght to be visible in graph
+    if (barLength < 40) {
+      barLength = 40;
+    }
+    const barHeight = 30;
+    const x = start[0];
+    const y = end[1];
+    const itemText = data[itemIndex as number].name;
+    const textWidth = echarts.format.getTextRect(itemText).width;
+    const text = barLength > textWidth + 40 && x + barLength >= 180 ? itemText : "";
+    const rectText = clipRectByRect(params, {
+      x: x,
+      y: y,
+      width: barLength,
+      height: barHeight
+    });
+
+    return {
+      type: "group",
+      // children due to have a possibility to have more different rectangles
+      children: [
+        {
+          type: "rect",
+          ignore: !rectText,
+          shape: rectText,
+          style: {
+            fill: api.visual("color")
+          },
+          textConfig: {
+            position: "insideLeft"
+          },
+          textContent: {
+            type: "text",
+            style: {
+              text: text,
+              fontSize: 12,
+              textFill: "#fff"
+            }
+          }
+        }
+      ]
+    };
+  };
+
+  function clipRectByRect(params: any, rect: any) {
+    return echarts.graphic.clipRectByRect(rect, {
+      x: params.coordSys.x,
+      y: params.coordSys.y,
+      width: params.coordSys.width,
+      height: params.coordSys.height
+    });
+  }
+
+  const option: ReactEChartsProps["option"] = {
+    tooltip: {},
+    dataZoom: [
+      {
+        type: "slider",
+        filterMode: "weakFilter",
+        showDataShadow: false,
+        top: 15,
+        labelFormatter: ""
+      },
+      {
+        type: "inside",
+        filterMode: "weakFilter"
+      },
+      {
+        type: "slider",
+        yAxisIndex: 0,
+        zoomLock: true,
+        width: 0,
+        right: "8%",
+        top: 60,
+        bottom: 640 - gridMaxHeight(),
+        start: 0,
+        end: verticalAxisZoomMaxValue(),
+        handleSize: 0,
+        showDetail: false
+      }
+    ],
+    grid: {
+      height: gridMaxHeight(),
+      show: true
+    },
+    xAxis: {
+      type: isTimeIndexed() ? "time" : "value",
+      position: "bottom",
+      splitLine: {
+        show: true,
+        lineStyle: {
+          color: ["#D3D3D3"]
+        }
+      }
+    },
+    yAxis: {
+      data: categories,
+      show: false,
+      inverse: true,
+      min: 0,
+      max: categories.length
+    },
+    series: [
+      {
+        type: "custom",
+        renderItem: renderGanttItem,
+        itemStyle: {
+          opacity: 0.8
+        },
+        label: {
+          show: true
+        },
+        encode: {
+          x: [1, 2],
+          y: 0,
+          tooltip: [dimStart, dimEnd]
+        },
+        data: data
+      }
+    ]
+  };
+
+  useEffect(() => {
```

Comment:
This useEffect and the one below seem to be unnecessary?

Reply:
fixed